release/rancher/rancher.go

```diff
@@ -167,6 +173,98 @@ type Content struct {
 	KDMWithDev     []ContentLine
 }
 
+type ListBucketResult struct {
+	Contents []struct {
+		Key string `xml:"Key"`
+	} `xml:"Contents"`
+}
+
+type ArtifactsIndexContent struct {
+	GA         ArtifactsIndexContentGroup `json:"ga"`
+	PreRelease ArtifactsIndexContentGroup `json:"preRelease"`
+}
+
+type ArtifactsIndexContentGroup struct {
+	Versions map[string][]string `json:"versions"`
+	BaseURL  string              `json:"baseUrl"`
+}
+
+func GeneratePrimeArtifactsIndex(writeToPath string) error {
+	client := ecmHTTP.NewClient(time.Second * 15)
+	resp, err := client.Get(rancherArtifactsListURL)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return errors.New("unexpected status code on " + rancherArtifactsListURL + " expected 200, got " + strconv.Itoa(resp.StatusCode))
+	}
+	defer resp.Body.Close()
+	contentDecoder := xml.NewDecoder(resp.Body)
+	listBucket := new(ListBucketResult)
```

Comment:
new is t really used anymore. Favor "var listBucket type" and take the address of that where used.